deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/CheckDeployEnvStatusWorker.java

```diff
@@ -0,0 +1,127 @@
+                if (resultBean.getTotal() < 1) {
+                    LOG.warn("Could not find any previous succeeded deploy in env {}", envBean.getEnv_id());
+                }
+                for (DeployBean deploy : resultBean.getDeploys()) {
+                    if (deploy.getState() == DeployState.RUNNING) {
+                        runningDeploys += 1;
+                    }
+                    else if (deploy.getState() == DeployState.FAILING) {
+                        failingDeploys += 1;
+                    }
+                    else if (deploy.getState() == DeployState.SUCCEEDING) {
+                        succeedingDeploys += 1;
+                    }
+                    else if (deploy.getState() == DeployState.SUCCEEDED) {
+                        succeededDeploys += 1;
+                    }
+                    else if (deploy.getState() == DeployState.ABORTED) {
+                        abortedDeploys += 1;
+                    }
+                }
+                index += 1;
+                filterBean.setPageIndex(index);
+            } 
+        }
+        emitMetrics(DeployState.RUNNING.toString(), runningDeploys);
+        emitMetrics(DeployState.FAILING.toString(), failingDeploys);
+        emitMetrics(DeployState.SUCCEEDING.toString(), succeedingDeploys);
+        emitMetrics(DeployState.SUCCEEDED.toString(), succeededDeploys);
+        emitMetrics(DeployState.ABORTED.toString(), abortedDeploys);
+    }
+
+    private void emitMetrics(String status, int stateCount) throws Exception {
+        String sentence = "";
+        Socket clientSocket = null;
+        DataOutputStream outToServer = null;
+        try {
+            clientSocket = new Socket("127.0.0.1", METRICS_AGENT_PORT);
+            outToServer = new DataOutputStream(clientSocket.getOutputStream());
+            sentence = String.format("put %s %d %d %s", status, System.currentTimeMillis(), stateCount, " host=localhost");
+            outToServer.writeBytes(sentence + '\n');
+        } catch (Exception e) {
+            LOG.error("TCPClientForMetrics exception {} ", e.toString());
+        } finally {
+            outToServer.close();
+            clientSocket.close();
+        }
+    }
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        SchedulerContext schedulerContext;
+
+        try {
+            schedulerContext = context.getScheduler().getContext();
+        } catch (SchedulerException e) {
+            LOG.error("Cannot retrive job context!", e);
+            return;
+        }
+
+        ServiceContext serviceContext = (ServiceContext) schedulerContext.get("serviceContext");
+        environDAO = serviceContext.getEnvironDAO();
+        deployDAO = serviceContext.getDeployDAO();
+        utilDAO = serviceContext.getUtilDAO();
+
+        try {
+            LOG.info("Start CheckDeployEnvStatusWorker process...");
+            processDeploys();
```

Comment:
Good point, but I do have concern if someone needs to setContext instead of just get it, without these variables passing in here. set will not work.
Although set operation might not happen often, but still pass in here can avoid that issue.